Modules/socketmodule.c

```diff
@@ -2605,7 +2605,42 @@ sock_settimeout(PySocketSockObject *s, PyObject *arg)
         return NULL;
 
     s->sock_timeout = timeout;
-    if (internal_setblocking(s, timeout < 0) == -1) {
+
+    int block = timeout < 0;
+    /* Blocking mode for a Python socket object means that operations
```

Comment:
Hmm, I'm sure that comment can be made much shorter :-)

Reply:
Can you write your version?

Reply:
Sure:
```
/* When a socket socket has a non-infinite timeout,
 * its underlying fd is internally set to non-blocking.
 * This table summarizes [etc.]
 */
```

Reply:
Hm, that's too short :)  But I'll trim down my version.

Reply:
Pushed an update.